update_blocklist.nix

```
@@ -40,16 +40,43 @@ in
   # Create an ip set and add each ip to it one by one
 
   # IPv4 and IPv6 regex patterns with CIDR notation support - WARNING: might not be correct for all IPs (e.g. ignore valid ones or accept wrong ones), but seems to work fine
+  # Also supports hosts files, as well as plain lists of domains.
   ipv4_regex="^([0-9]{1,3}\.){3}[0-9]{1,3}(\/[0-9]{1,2})?$"
   ipv6_regex="^([0-9a-fA-F:]+::?[0-9a-fA-F]*)+(\/[0-9]{1,3})?$"
+  domain_regex="^[a-zA-Z0-9.-]*[a-zA-Z][a-zA-Z0-9.-]*$"
```

What do you think about "at least one literal `.`"? That would prohibit things like `localhost`. Not sure if that is a good or a bad thing 🤔 If so it could be
```suggestion
  domain_regex='^[a-zA-Z0-9.-]*\.[a-zA-Z][a-zA-Z0-9.-]+$'
```

---

I didn't think about it, but there are [CNAME records](https://en.wikipedia.org/wiki/CNAME_record) as well and apparently dig is querying them, too by default:
> dig @127.0.0.1 mclean.f.360.cn A +short
```
i18nv2.mclean.50union.com.
8.222.197.237
104.192.110.73
8.208.86.177
8.208.86.166
```
Indeed we _should_ deal with them, too, since otherwise the blocklist would not work
So basically we need to do the regex match on the results and call `blockDomain` recursively

---

And it isn't dig, it is the DNS server: https://unix.stackexchange.com/a/723316
So no matter if we do separate dig invocations or a single command, we will always get duplicate CNAME information from simply querying A and AAAA record… Not really nice, but the script could deal with it and just ignores duplicate IPs (That is the `-exist` flag for ipset)

---

Alright, I think it may be a good thing.

Users wouldn't want strange errors from not being able to connect to local servers (as some services require.)

I think it would be a good thing.